packages/job-worker/src/cache/CacheCollection.ts

```diff
@@ -69,42 +63,13 @@ export class DbCacheReadCollection<TDoc extends { _id: ProtectedString<any> }> {
 		return this._collection.name
 	}
 
-	findFetch(selector: MongoQuery<TDoc> | TDoc['_id'] | SelectorFunction<TDoc>, options?: FindOptions<TDoc>): TDoc[] {
+	findFetch(selector: SelectorFunction<TDoc> | null, options?: FindOptions<TDoc>): TDoc[] {
```

Comment:
I wonder, if we're changing the signatures this much, and are actively going away from pretending to be Mongo, should this be called just `find` or `findAll`, so that no suggestion of this being a Meteor Cursor or a Mongo query is made?

Reply:
Also, I suppose the meaning of `null` should be clarified through some JSDoc, so that it can be intellisensed where it's being used.

Reply:
yeah `findAll` would make sense
docs probably do need some work..

I was wondering about changing the structure of the options parameter too, as some bits it can do really arent that useful..
eg, we dont clone the objects except if a `fields` property is defined. and as there is no reactivity, we dont need to use it to limit that. so `fields` adds a cost than an optimization here (the opposite normal mongo queries).
And i think the sorting algorithm is fairly primitive/inefficient
But I'm leaving that for a follow up